### PR TITLE
Typescriptify config module

### LINKED ISSUE #50: db connection boilerplate
Setup the db connection/config, and any other minor miscellany.

### LINKED ISSUE #52: Update config module to use typescript

### PR DESCRIPTION
**Pull request by eliascodes**
Fixes #52 
Fixes #50 

The size of this PR is artificially inflated b/c of the package-lock and the fact that git thinks some files have been removed/added instead of moved + changed.

Changes:
* Move the config module to typescript
* Add stub for api plugin
* Minor re-organisation of server setup files
* Add setup of db connection

### CODE REVIEW

**Comment by astroash** [comment]
@eliascodes have you rebased master?

**Comment by eliascodes** [comment]
yes

File: src/types/internal.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export type Nothing = null;
+export type Maybe<T> = T | Nothing;
+
+export type Map<K extends string | number, V> = {
+  [k in K]: V
+}
+
+export type Dictionary<T> = {
```
Comment (astroash) [inline src/types/internal.d.ts]: Now that we have a generic types folder, do you mind importing it here:

https://github.com/TwinePlatform/twine-api/blob/cbe6de1efff3c3a2bc2237bf9b19b32fbd0d66aa/src/auth/permissions/permissions.ts#L3

Comment (eliascodes) [reply src/types/internal.d.ts]: Done

**Pull request merged**